### PR TITLE
Currency API integration and shifting some files around

### PR DESCRIPTION
**Pull request by tanadendere**
Changes made:

- Integrated with the currency API 
- Added a currency selection for user to select their preferred currency
- Made a feature to calculate the total cost of the trip

### CODE REVIEW

File: src/app/tripManagement/components/trip-details/itinerary-item/itinerary-item.component.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export class ItineraryItemComponent {
   trip: ITrip | undefined = undefined;
   tripSubscription = new Subscription();
 
-  ngOnInit() {
+  constructor() {
     this.tripSubscription = this.trip$.subscribe((trip) => {
```
Comment (Musa-Mabasa) [inline src/app/tripManagement/components/trip-details/itinerary-item/itinerary-item.component.ts:26]: Just a suggestion. Another way to unsubscribe would be to use `takeUntilDestroyed` from `rxjs-interop` in this way: `this.trip$.pipe(takeUntilDestroyed()).subscribe{...}`. This is, however, just a suggestion, what you have is also  correct.

File: src/app/tripManagement/components/trip-details/trip-details.component.html [hunk 2]
```diff
@@ -126,6 +128,10 @@ <h2>Add to your itinerary</h2>
       formControlName="endDateTime"
     />
   </div>
+  @if (selectedCurrency$ | async; as selectedCurrency) {
+  <div>{{ selectedCurrency.symbol_native }}</div>
```
Comment (Musa-Mabasa) [inline src/app/tripManagement/components/trip-details/trip-details.component.html]: If this object is not from a library, evething `TS` should use `camelCase`.

File: src/app/tripManagement/utilities/utils.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { IExchangeRateData, IExchangeRate } from '../models/exchangeRate';
+import { IItineraryItem } from '../models/itinerary';
+
+export function getCurrencyCodes(itinerary: IItineraryItem[]) {
+  let currencyCodes: string[] = [];
```
Comment (Musa-Mabasa) [inline src/app/tripManagement/utilities/utils.ts]: You can use `const` for this variable since you are not changing it's reference.

**Review by Musa-Mabasa — CHANGES_REQUESTED** [review]
I like the use of observables on this one. My only request is the naming convention for `symbol_native` and similar properties. Either than that, well done 👍

File: src/app/tripManagement/components/trip-details/itinerary-item/itinerary-item.component.ts (hunk 1 above)
Comment (tanadendere) [reply src/app/tripManagement/components/trip-details/itinerary-item/itinerary-item.component.ts:26]: Thank you, yea Hugo also suggested the same too. I've noted it and will get to it if i have time to polish the code at the end

File: src/app/tripManagement/components/trip-details/trip-details.component.html [hunk 4]
```diff
@@ -129,7 +129,7 @@ <h2>Add to your itinerary</h2>
     />
   </div>
   @if (selectedCurrency$ | async; as selectedCurrency) {
-  <div>{{ selectedCurrency.symbol_native }}</div>
+  <div>{{ selectedCurrency.symbolNative }}</div>
```
Comment (Musa-Mabasa) [inline src/app/tripManagement/components/trip-details/trip-details.component.html:132]: Nice!!

File: src/app/userManagement/models/currency.d.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
   symbol: string;
   name: string;
-  symbol_native: string;
-  decimal_digits: number;
+  symbolNative: string;
+  decimalDigits: number;
   rounding: number;
   code: string;
-  name_plural: string;
+  namePlural: string;
```
Comment (Musa-Mabasa) [inline src/app/userManagement/models/currency.d.ts:8]: Very nice!!

**Review by Musa-Mabasa — APPROVED** [review]
Everything looks good, very good 👍

**Review by AyaazStreet — APPROVED** [review]
Everything seems right just maybe the things Musa mentioned. Good job

**Pull request merged**